### PR TITLE
Updates to support configurable URLs to our backend

### PR DESCRIPTION
**Pull request by doanac**
These changes allow us to run CI jobs against services not under foundries.io.

### CODE REVIEW

File: helpers.sh
Comment (mike-sul) [inline helpers.sh:35]: Maybe we should do it the container image during its building?

File: apps/build.sh
Comment (mike-sul) [inline apps/build.sh:118]: Where `hub_fio` is defined? In the login function. Maybe we should introduce some env. var for it.

File: apps/build.sh
Comment (mike-sul) [inline apps/build.sh:66]: I guess it cannot be specified via a CLI param?

File: apps/build.sh
Comment (mike-sul) [inline apps/build.sh:63]: I suppose this file can be generating during the container image building (lmp-inage-tools).

File: helpers.sh
Comment (mike-sul) [inline helpers.sh:15]: Why not use `FIO_HUB_URL` here?

File: lmp/bb-config.sh
Comment (mike-sul) [inline lmp/bb-config.sh:15]: Maybe we should do the same for apps/publish?

**Review by mike-sul — APPROVED** [review]
LGTM as long as we are fine with hard-coding the domain prefixes (api, hub, etc).

File: apps/build.sh
Comment (doanac) [reply apps/build.sh:118]: the docker-login function sets this as a variable

File: apps/build.sh
Comment (doanac) [reply apps/build.sh:66]: Not according to what I read

File: apps/build.sh
Comment (doanac) [reply apps/build.sh:63]: I don't think we want a custom image per instance.

File: helpers.sh
Comment (doanac) [reply helpers.sh:15]: it would be the other way around, but FIO_HUB_URL is set in bb-config.sh. Its only available to a yocto build. this is getting setting for container builds.

File: lmp/bb-config.sh
Comment (doanac) [reply lmp/bb-config.sh:15]: i think apps/publish handles this also?

File: apps/build.sh
Comment (mike-sul) [reply apps/build.sh:63]: Is this directory `/usr/local/share/ca-certificates` of container mapped to a host directory?

File: apps/build.sh
Comment (mike-sul) [inline apps/build.sh:65]: Ah., this is an instance specific.

File: apps/build.sh
Comment (doanac) [reply apps/build.sh:63]: exactly - https://github.com/foundriesio/jobserv/commit/e7affa0eae6ae7ae8248bdb7cc7a8842f9430b44

File: apps/build.sh
Comment (mike-sul) [reply apps/build.sh:66]: `ca=["/etc/ssl/certs/ca-certificates.crt"]` but we check `if [ -d /usr/local/share/ca-certificates ]`, I mean we check different directory than use?

Comment (doanac) [reply apps/build.sh:66]: the `load_extra_certs` takes the certs and produces an updated `/etc/ssl/certs/ca-certificates.crt` - so this more about making the container use the same certs as this "host"

Comment (mike-sul) [reply apps/build.sh:66]: ok, a bit subtle.

File: helpers.sh
Comment (mike-sul) [reply helpers.sh:15]: I understand it. My point is that defining the env. var. at the beginning of the overall script (maybe publish.sh) is less subtle then defining `hub_fio` in one of the functions and then using it at the other places.

File: lmp/bb-config.sh
Comment (mike-sul) [reply lmp/bb-config.sh:15]: I just thought about adding those env. vars to `publish.sh`.

**Comment by mike-sul** [comment]
LGTM. Those comments are not important. It would be great to fix the unit tests too.

File: helpers.sh
Comment (doanac) [reply helpers.sh:15]: we've never defined variable globally in something like helpers.sh so that multiple scripts get a value. We could do something like the `dns_base` function in helpers.sh though. Then everyone has access to it.

Comment (doanac) [reply helpers.sh]: I've cleaned this up a little bit and just set dns_base and hub_fio as environment variables. gonna test this stuff out and then look at the unit-test

**Comment by doanac** [comment]
@mike-sul - unit tests are passing and its working in meds and my andy-corp factory. Care to give this a spin before I merge?

File: lmp/bb-config.sh
Comment (mike-sul) [inline lmp/bb-config.sh:56]: Maybe if we set `H_RUN_URL=${H_RUN_URL}` in `conf/local.conf` then we will not need this dynamic scripting? Similar case (`GARAGE_TARGET_VERSION = "${H_BUILD}"` at #94.

Comment (doanac) [reply lmp/bb-config.sh:56]: you are probably right. @ricardosalveti - is mike's suggestion preferable for this?

Comment (mike-sul) [reply lmp/bb-config.sh:56]: I think it's not so important to be a blocker for //M.

Comment (quaresmajose) [reply lmp/bb-config.sh:56]: IMO we should avoid using `mktemp` in bitbake variable because it breaks the sstate cache, they generate a different values in each run. Something like `${HERE}/fetch-root-meta-helper.sh` look better to me.

This variable seems to only affect the tasks of the final target image that always runs on the CI build (do not use sstate-cache) but avoid the `mktemp` usage on variable names is good police in my opinion.

Comment (quaresmajose) [reply lmp/bb-config.sh:56]: The script to looks good to me, it's just the name that worried me because they are not reproducible

Comment (doanac) [reply lmp/bb-config.sh]: rebased on latest master and then fixed in force-push

**Pull request merged**
